src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -103,69 +72,68 @@ public Utf8String ReadString()
             return str;
         }
 
-        public ValueType GetValueType()
+        public JsonValueType GetJsonValueType()
         {
             SkipEmpty();
-            var nextByte = _str[_index];
+            var nextByte = (byte) _str[_index];
 
-            if (nextByte == CurlyOpenString)
+            if (nextByte == '{')
             {
-                return ValueType.Object;
+                return JsonValueType.Object;
             }
 
-            if (nextByte == SquareOpenString)
+            if (nextByte == '[')
             {
-                return ValueType.Array;
+                return JsonValueType.Array;
             }
 
-            if (nextByte == DashString || (nextByte.Value >= ZeroString.Value && nextByte.Value <= NineString.Value))
+            if (nextByte == '-' || (nextByte >= '0' && nextByte <= '9'))
             {
-                return ValueType.Number;
+                return JsonValueType.Number;
             }
 
-            if (nextByte == StartTrueString)
+            if (nextByte == 't')
             {
-                return ValueType.True;
+                return JsonValueType.True;
             }
 
-            if (nextByte == StartFalseString)
+            if (nextByte == 'f')
             {
-                return ValueType.False;
+                return JsonValueType.False;
             }
 
-            if (nextByte == StartNullString)
+            if (nextByte == 'n')
             {
-                return ValueType.Null;
+                return JsonValueType.Null;
             }
 
-            if (nextByte == QuoteString)
+            if (nextByte == '"')
             {
-                return ValueType.String;
+                return JsonValueType.String;
             }
 
             throw new ArgumentOutOfRangeException();
```

Comment:
Agreed.
Although, I was considering changing the API to take in an argument:
public Utf8String GetValue(JsonValueType type)